src/layers/stateless.jl

```diff
@@ -5,56 +5,81 @@ using NNlib: logsoftmax, logσ
 
 mse(ŷ, y) = sum((ŷ .- y).^2) * 1 // length(y)
 
-function _crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, weight::Nothing)
-  return -sum(y .* log.(ŷ)) * 1 // size(y, 2)
-end
+_crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, weight::Nothing) =
+  -sum(y .* log.(ŷ)) * 1 // size(y, 2)
 
-function _crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, weight::Number)
-  return -sum(y .* log.(ŷ)) .* weight * 1 // size(y, 2)
-end
+_crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, weight::Union{Number, AbstractVector}) =
+  _crossentropy(ŷ, y.*weight, nothing)
 
-function _crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, weight::AbstractVector)
-  return -sum(y .* log.(ŷ) .* weight) * 1 // size(y, 2)
+function crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat; weight=nothing, label_smoothing=zero(eltype(y)))
+  if !iszero(label_smoothing)
+    y = y .* (1 - label_smoothing) .+ label_smoothing* 1 // size(y, 1)
```

Comment:
Is it worth being explict:
```suggestion
    y = y .* (1 - label_smoothing) .+ (label_smoothing // size(y, 1))
```

Reply:
Yes. label_smoothing can be non-integral type that the operator `//` may not support.

Reply:
Ah..
```suggestion
    y = y .* (1 - label_smoothing) .+ (label_smoothing * 1 // size(y, 1))
```